ivy/data_classes/array/array.py

```diff
@@ -188,28 +189,27 @@ def dynamic_backend(self):
 
     @dynamic_backend.setter
     def dynamic_backend(self, value):
-        from ivy.functional.ivy.gradients import _variable, _is_variable, _variable_data
-        from ivy.utils.backend.handler import _determine_backend_from_args
+        from ivy.functional.ivy.gradients import _variable
+        from ivy.utils.backend.handler import _data_to_new_backend
 
-        if value == False:
-            self._backend = _determine_backend_from_args(self).backend
-
-        else:
+        if value:
             ivy_backend = ivy.with_backend(self._backend)
-            to_numpy = ivy_backend.to_numpy
 
-            if _is_variable(self.data) and self._backend not in ["jax", "numpy"]:
-                native_data = _variable_data(self.data)
-                np_data = to_numpy(native_data)
-                new_arr = ivy.array(np_data)
-                self._data = _variable(new_arr).data
+            if ivy_backend.gradients._is_variable(self.data):
+                native_var = ivy_backend.gradients._variable_data(
+                    self,
+                )
+                data = _data_to_new_backend(native_var, ivy_backend).data
+                self._data = _variable(data).data
```

Comment:
ig this could also be `ivy_backend.gradients._variable` instead?

Reply:
`ivy_backend` is the old backend, but we need to call this function using `ivy` which uses the new backend

Reply:
got it! Thanks for clarifying